drivers/spi/spi.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2023 Graphcore Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "spi.h"
+
+bool spi_buf_set_in_nocache(const struct spi_buf_set *bufs)
+{
+	for (size_t i = 0; i < bufs->count; i++) {
+		const struct spi_buf *buf = &bufs->buffers[i];
+
+		if (!dt_buf_in_nocache((uintptr_t)buf->buf, buf->len)) {
+			return false;
+		}
+	}
+	return true;
+}
```

Comment:
> can you make this static inline and move this function into include/zephyr/drivers/spi.h so to not have at all spi.c and spi.h from this PR?

It was static inline before and I moved it because it was requested in comments above, and I agreed as I prefer to have implementations in C files since tests are simpler.

Reply:
agree with @carlocaione this should be in spi.h as a static inline